torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -22,6 +22,31 @@ namespace rpc {
 
 namespace {
 
+void DeleteFutureIValue(FutureIValue* fv) {
+  pybind11::gil_scoped_acquire ag;
+  delete fv;
+}
+
+std::shared_ptr<FutureIValue> toFutureIValue(
```

Comment:
There are 2 definitions of `toFutureIValue()`, can we possibly dedupe these? Is there guidance on which API to use?

Reply:
Oh, the one in `python_function.cpp` assumes the IValue contains a `py::object`, and the one in `py_rref.cpp` does not, it is only for signaling the completion. Yes, let me consolidate these two into a util function.

Reply:
If it's tricky to dedupe because of GIL, custom deleter, etc maybe rename such as `toFutureIValue` and `toFutureIValuePyObj` or something of the like

Reply:
This should be OK fo now, as both scenarios are Python-only (`python_function.cpp` and `py_rref.cpp`). Will check if this can be an issue when we merge this with `ivalue::Future`